File: 0989-add-to-array-form-of-integer/0989-add-to-array-form-of-integer.py
class Solution:
    def addToArrayForm(self, num: List[int], k: int) -> List[int]:
        
        temp = 0
        for i in num:
            temp = temp * 10 + int(i) 
        
        temp = temp + k
        
        temp = str(temp)
       
        ans = []
        
        for i in range(len(temp)):
            
            ans.append(int(temp[i]))
        
        return ans

File: 0989-add-to-array-form-of-integer/0989-add-to-array-form-of-integer.cpp
class Solution {
public:
    vector<int> addToArrayForm(vector<int>& num, int k) {
        
        int N = num.size();
        for(int i = N - 1 ; i >= 0 ; i--)
        {
            int temp = (num[i] + k);
            num[i] = temp % 10;
            k = (temp) / 10;
        }
        
        while(k > 0)
        {
            num.insert(num.begin() , k % 10);
            k = k / 10;
        }
        
        return num;
    }
};